flake.nix

```diff
@@ -0,0 +1,109 @@
+{
+  description = "A very basic flake";
+
+  nixConfig = {
+    flake-registry = "https://github.com/serokell/flake-registry/raw/master/flake-registry.json";
+  };
+
+  inputs = {
+    haskell-nix.url = "github:input-output-hk/haskell.nix";
+    nixpkgs.url = "github:serokell/nixpkgs";
+    hackage.flake = false;
+    stackage.flake = false;
+  };
+
+  outputs = { self, nixpkgs, flake-utils, haskell-nix, hackage, stackage, ... }@inputs:
+  flake-utils.lib.eachSystem ["x86_64-linux"] (system: let
+    pkgs = import inputs.nixpkgs {
+      inherit system;
+      overlays = [ inputs.haskell-nix.overlay ];
+    };
+
+    # run ci with resolver defined in stack.yaml and a newer resolver for servant 0.19
+    resolvers = [ null "lts-18.28" ];
+
+    local-packages = [
+      { name = "servant-util";         subdirectory = "servant-util"; }
+      { name = "servant-util-beam-pg"; subdirectory = "servant-util-beam-pg"; }
+    ];
+
+    # names of all local packages
+    local-packages-names = map (p: p.name) local-packages;
+
+    make-hs-pkgs = resolver: let
+      # source with gitignored files filtered out
+      projectSrc = pkgs.haskell-nix.haskellLib.cleanGit { name = "servant-util"; src = ./.; };
+
+      # haskell.nix does not support 'include' in package.yaml, we have to generate .cabal ourselves
+      cabalFiles = pkgs.runCommand "servant-util-cabal-files" {} ''
+        mkdir -p $out
+        ${pkgs.lib.concatMapStrings ({ name, subdirectory }: ''
+          ${pkgs.haskellPackages.hpack}/bin/hpack ${projectSrc}/${subdirectory} - > $out/${name}.cabal
+        '') local-packages}
+      '';
+
+      cabalsrc = pkgs.runCommand "src-with-cabal" {} ''
+        cp -r --no-preserve=mode ${projectSrc} $out
+        ${pkgs.lib.concatMapStrings ({ name, subdirectory }: ''
+          cp ${cabalFiles}/${name}.cabal $out/${subdirectory}
+        '') local-packages}
+      '';
+
+      src = (if resolver == null then cabalsrc else let
+        hlib = pkgs.haskell-nix;
+        allSnapshots = builtins.attrNames hlib.snapshots;
+        isLts = version: builtins.substring 0 3 version == "lts";
```

Comment:
Is it used?

Reply:
Meant to put in a helpful assert but hackage.nix and stackage.nix have decent enough error messages, removed!